sigstore/_internal/oidc/issuer.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Helper that queries the OpenID configuration for a given issuer and extracts its endpoints
+"""
+
+import urllib.parse
+
+import requests
+
+
+class IssuerError(Exception):
+    pass
+
+
+class Issuer:
+    def __init__(self, base_url: str) -> None:
+        oidc_config_url = urllib.parse.urljoin(
+            f"{base_url}/", ".well-known/openid-configuration"
+        )
+
+        resp: requests.Response = requests.get(oidc_config_url)
+        try:
+            resp.raise_for_status()
+        except requests.HTTPError as http_error:
+            raise IssuerError from http_error
+
+        self.struct = resp.json()
```

Comment:
`self._struct` perhaps? Alternatively, if we don't need any other fields from the OpenID configuration payload, it might be worth not binding this to `self`.

Reply:
[50e342b](https://github.com/sigstore/sigstore-python/pull/62/commits/50e342b05e24a7d52e5f2fd9254ec357b3647682)